src/Controller/AdminController.php

```diff
@@ -24,15 +29,84 @@
 class AdminController extends EasyAdminController
 {
     private $userManager;
+    private $newMaterialService;
 
     /**
      * AdminController constructor.
      *
      * @param UserManagerInterface $userManager
+     * @param NewMaterialService   $newMaterialService
      */
-    public function __construct(UserManagerInterface $userManager)
+    public function __construct(UserManagerInterface $userManager, NewMaterialService $newMaterialService)
     {
         $this->userManager = $userManager;
+        $this->newMaterialService = $newMaterialService;
+    }
+
+    public function queryBatchAction(array $ids)
+    {
+        $ids = array_map(static function ($id) {
+            return (int) $id;
+        }, $ids);
+        $searches = $this->em->getRepository(Search::class)->findBy(['id' => $ids]);
+
+        $date = new \DateTimeImmutable('7 days ago');
```

Comment:
Hardcoded "7 days ago" ?

Reply:
Should be given as a get parameter. But if we do that we should also paginate, so ignore for now

Reply:
Added todo